src/main/java/dev/mcnees/moneymapper/MoneyMapperRunner.java

```diff
@@ -0,0 +1,97 @@
+package dev.mcnees.moneymapper;
+
+import java.io.File;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.sql.DataSource;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.batch.core.Job;
+import org.springframework.batch.core.JobParameter;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.JobParametersBuilder;
+import org.springframework.batch.core.launch.JobLauncher;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MoneyMapperRunner implements ApplicationRunner {
+
+	private static final Logger log = LoggerFactory.getLogger(MoneyMapperRunner.class);
+
+	private final JobLauncher jobLauncher;
+
+	private final Job moneyMapperJob;
+
+	private final JdbcTemplate jdbcTemplate;
+
+	@Value("${qfx_directory:qfx_files}")
+	private String qfx_directory_parameter;
+
+	@Value("${output_directory:output}")
+	private String output_directory_parameter;
+
+	public MoneyMapperRunner(JobLauncher jobLauncher, Job moneyMapperJob, DataSource dataSource) {
+		this.jobLauncher = jobLauncher;
+		this.moneyMapperJob = moneyMapperJob;
+		jdbcTemplate = new JdbcTemplate(dataSource);
+	}
+
+	@Override
+	public void run(ApplicationArguments args) throws Exception {
+
+		clearProcessingData();
+
+		File qfx_directory = new File(qfx_directory_parameter);
+		if (!qfx_directory.isDirectory()) {
+			log.error("Specified qfx input location of " + qfx_directory_parameter + " is not a directory");
+			throw new Exception("Must specify a directory for QFX file processing.");
+		}
+
+		File output_directory = new File(output_directory_parameter);
+		if (!output_directory.isDirectory()) {
+			log.error("Specified output location of " + output_directory_parameter + " is not a directory");
+			throw new Exception("Must specify a valid file for output.");
+		}
+		File output_file = new File(output_directory + "/money_mapper_output.csv");
+
+		List<File> allQuickenFiles = getAllQuickenFiles(qfx_directory);
+
+		JobParameters jobParameters;
+		for (File quickenFile : allQuickenFiles) {
+			jobParameters = new JobParametersBuilder()
+					.addJobParameter("input_file", new JobParameter<>(quickenFile, File.class))
+					.addJobParameter("output_file", new JobParameter<>(output_file, File.class))
+					.addJobParameter("datetime", new JobParameter<>(Instant.now().toEpochMilli(), Long.class))
+					.toJobParameters();
+
+			jobLauncher.run(moneyMapperJob, jobParameters);
```

Comment:
I remember this now.  I started with this approach initially and got stuck.  I was new(er) to Batch and didn't know how to handle 2 delegates.  I'm still not sure I do.

The `MultiResourceItemReader` uses a delegate to do the actual reading.  The problem is that each file needs a [multi-line reader ](https://docs.spring.io/spring-batch/docs/current/reference/html/common-patterns.html#multiLineRecords) which also uses a delegate.  I didn't know how to reconcile 2 delegates in a single reader.  This is when I talked to Mahmoud and he recommended using an `ApplicationRunner` with `JobParameters`.

I suppose I could create a `MultiResourceItemReader` and pass my custom `MultilineQFXReader` as an `ItemStreamReader` to `setDelegate`.  My `MultilineQFXReader` would still have it's own delegate.  Is that how you would see the nested delegates?

Reply:
Yes. I would expect the delegate for the `MultiResourceItemReader` to be your `MultilineQFXReader` (although that reader will need to also implement `ResourceAware`). The delegate for the `MultilineQFXReader` would be the `FlatFileItemReader` as you have already done. Turtles all the way down ;)

Reply:
I gave this a try and quickly realized that it only takes a few turtles for me to get lost.  I'm remembering all of my old problems and questions real quick.
For example, I can create the `MultiResourceItemReader` with few problems.  I can assign a `Resource` and I can also assign a new `ResourceAwareItemReaderItemStream` (`MultilineQFXReader` in this case) as the delegate.
The problem I'm having is that the `MultilineQFXReader` also has a delegate that needs a `Resource` to represent each file to process.  So I need the output from `MultiResourceItemReader` to feed into the delegate's delegate.  My current code injects the file `Resource` as a constructor arg in `MultilineQFXReader`, so that needs to change somehow.  I'm trying to create a new `MultilineQFXReader` with a Resource that needs to be determined by the `MultiResourceItemReader`.  However the `MultiResourceItemReader` doesn't exist without `MultilineQFXReader` as a delegate but the `MultilineQFXReader` doesn't exist until `MultiResourceItemReader` gives it the right file and on and on the circle goes.

All that to say I've thoroughly confused myself again and I need to understand the delegate pattern a bit more.  I'll be tackling this change a bit later.

Reply:
I tried this implementation and quickly hit the same problems I had the first time around.  I remember it quite well now and how long I got stuck for.

I don't think I can succinctly describe my problem, which to me indicates a lack of true understanding on my part particularly of the delegate pattern.  I'll briefly try to explain one problem but I doubt this will go well.

In my mind it comes down to creation order.  I can create a `MultiResourceItemReader` no problem, but the delegate I use is `MultilineQFXReader` which has its own delegate and needs its own `Resource`.  So I can't create a `MultilineQFXReader` until I have a file `Resource` that will be derived from `MultiResourceItemReader`, but I can't create the `MultiResourceItemReader` to get that file because I need the `MultilineQFXReader` to exist as a delegate.

I'm not likely to be implementing this change anytime soon.  I need to get a much better understanding of the delegate design pattern and the Batch classes involved before I try again.  Something feels fundamentally off in my understanding and anything I build at this point feels like a hack.  Back to the classroom first.